cpp/include/cugraph/algorithms.hpp

```diff
@@ -1583,7 +1583,7 @@ uniform_random_walks(raft::handle_t const& handle,
                      std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
                      raft::device_span<vertex_t const> start_vertices,
                      size_t max_length,
-                     uint64_t seed = std::numeric_limits<uint64_t>::max());
+                     raft::random::RngState& rng_state);
```

Comment:
Not a request for a change, mainly a comment for @ChuckHastings 

`rng_state` is mostly a parameter before `graph_view` in `algorithms.hpp` (exceptions, `maximal_independent_set` and `vertex_coloring`). In `sampling.hpp`, `rng_state` appears close to the end of the input parameter lists for uniform and biased sampling functions. We should better be consistent on this, but not sure placing this right before `graph_view` (or right after `handle`) make sense or not. This may make sense if we consier `rng_state` (sort of a common resource) as an extension of `handle`.

Reply:
I agree that consistency is better.

I'm inclined to prefer right after the handle.  It does seem like the RNG state is a common resource/extension of `handle` more than a regular parameter.  I think we have an optional case as well (Louvain/Leiden).  If we try and put it anywhere else in the argument list it seems like it could be pretty variable (e.g. before or after properties, etc).

Reply:
Let's move the rng_state parameter (per the discussion above) so that it becomes the second parameter (between the handle and the graph view).  That will necessitate updating the implementation and any tests or other calls to this function.